### ISSUE TITLE
__slots__ check gives false positive for inherited descriptor stored as class variable when subclass has own slots

### ISSUE THREAD

**Issue by jab** [microsoft/pyright#3183]
```python
from typing import Any


class MyDescriptor:
    def __set__(self, instance: object, value: object) -> None: ...
    def __get__(self, instance: object, owner: Any) -> Any: ...


class Node:
    __slots__ = ('nxt',)
    prv = MyDescriptor()


class SentinelNode(Node):
    __slots__ = ('foo',)  # this confuses pyright
    nxt = MyDescriptor()

    def repro(self) -> None:
        self.prv = Node()  # false positive occurs here
```

```console
❯ pyright test.py
No configuration file found.
No pyproject.toml file found.
stubPath /private/tmp/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
/private/tmp/test.py
  /private/tmp/test.py:19:14 - error: "prv" is not specified in __slots__ (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
Completed in 0.604sec
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.232, which I just published. It will also be included in the next release of pylance.

**Comment by jab** [comment]
Confirmed I no longer hit this after upgrading. Thanks!
